### PR TITLE
stripped issuer from label and updated unit tests

### PR DESCRIPTION
**Pull request by shivasheeshyadav**
Fix for issue #551

Issuer is now stripped from the label 

Tests are updated accordingly.

### CODE REVIEW

File: app/src/main/java/org/shadowice/flocke/andotp/Database/Entry.java [hunk 1]
```diff
@@ -131,6 +131,9 @@ public Entry(String contents) throws Exception {
 
         String secret = uri.getQueryParameter("secret");
         String label = uri.getPath().substring(1);
+        if(label.contains(":")) {
+            label = label.split(":")[1];
```
Comment (flocke) [inline app/src/main/java/org/shadowice/flocke/andotp/Database/Entry.java:135]: I think only splitting the label if it contains a `:` is not the best way of doing this. There could potentially be cases where it contains an `:` without the issuer.
We should probably check if it starts with `<issuer>:` and remove that part specifically.

Comment (shivasheeshyadav) [reply app/src/main/java/org/shadowice/flocke/andotp/Database/Entry.java:135]: @flocke Thanks for the suggestions. I'll make a commit towards this requested change.

Comment (SagePtr) [reply app/src/main/java/org/shadowice/flocke/andotp/Database/Entry.java:135]: Here is how it's done in the Google Authenticator:
https://github.com/google/google-authenticator-android/blob/master/java/com/google/android/apps/authenticator/otp/AccountDb.java#L163

Comment (shivasheeshyadav) [reply app/src/main/java/org/shadowice/flocke/andotp/Database/Entry.java]: @flocke @SagePtr
Please review the recent changes

File: app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java [hunk 2]
```diff
@@ -58,12 +60,13 @@
 import javax.crypto.spec.IvParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
 
-public class ApplicationTest extends ApplicationTestCase<Application> {
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
```
Comment (flocke) [inline app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java:64]: Use

`import static org.junit.Assert.assertEquals;`
`import static org.junit.Assert.assertTrue;`

instead.

File: app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java [hunk 3]
```diff
@@ -23,14 +23,16 @@
 
 package org.shadowice.flocke.andotp;
 
-import android.app.Application;
 import android.content.Context;
-import android.test.ApplicationTestCase;
+import androidx.test.InstrumentationRegistry;
```
Comment (flocke) [inline app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java:27]: Deprecated, use `androidx.test.platform.app.InstrumentationRegistry` instead.

File: app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java [hunk 4]
```diff
@@ -168,22 +174,29 @@ public void testEntryURL() throws Exception {
...
         assertEquals("HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ", new String(new Base32().encode(entry.getSecret())));
         String[] tags = new String[]{"test1", "test2"};
         assertEquals(tags.length, entry.getTags().size());
         assertTrue(Arrays.equals(tags, entry.getTags().toArray(new String[entry.getTags().size()])));
     }
 
+    @Test
     public void testSettingsHelper() throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
-        Context context = getContext();
+        Context context = InstrumentationRegistry.getTargetContext();
```
Comment (flocke) [inline app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java:199]: This is deprecated, use `InstrumentationRegistry.getInstrumentation().getTargetContext();` from `androidx.test.platform.app.InstrumentationRegistry` (see comment above).

**Comment by shivasheeshyadav** [comment]
> The stripping of the label is now ok, but there are still a lot of problems with the tests. I added some comments how I got them to work on my system (they didn't work with just your PR).

Thanks @flocke 
I've fixed the tests as per your suggestions. Please review.

**Pull request merged**